### DISCUSSION TITLE
For HNSW,ef_construction and M range in v2.4.0

### DISCUSSION THREAD

**Discussion by skylersky** [milvus-io/milvus#32751] [Q&A and General discussion]
I was testing with different M and ef_construction values for my project, and it seems the range for efConstruction:[8, 512], M: [4,64]. For version 2.4.0. 
![image](https://github.com/milvus-io/milvus/assets/149104425/3f45e69b-862e-48f6-b6d8-b3fd7a3f0325)

**Comment by yhmo** [comment]
I didn't see any change about parameter range between 2.3 and 2.4 in the online doc:
https://milvus.io/docs/v2.3.x/index.md#HNSW
https://milvus.io/docs/index.md#HNSW

Both of the docs show the same range:
M     (2, 2048)
efConstruction        (1, int_max)

**Reply by skylersky** [reply]
yes, in the docu it's same as what you said. I mean, when i set M = 100, the output error said M out of range [4,64], 
![image](https://github.com/milvus-io/milvus/assets/149104425/3438b592-9128-480a-97c0-3d8a698bbb3e)
same thing happen when i set efConstruction = 600 when build the collection
![image](https://github.com/milvus-io/milvus/assets/149104425/5ab64c66-32c3-4d9e-8ee8-309c48bd2f28)

**Reply by yhmo** [reply]
Which version of your milvus server?
I guess your python sdk version is v2.4.0, but the Milvus server version is v2.2.x.

In Milvus v2.2.x, the range of M is [4, 64].
https://milvus.io/docs/v2.2.x/index.md

**Reply by yhmo** [reply]
Print the server version:
```
connections.connect(host='localhost', port='19530')
print(utility.get_server_version())
```

**Reply by skylersky** [reply]
![image](https://github.com/milvus-io/milvus/assets/149104425/1dbbf397-29e6-42ed-b3e9-5228bab26778)

**Comment by xiaofan-luan** [comment]
I double checked the code, the M check is 2-2048.
could you share the stack of the error and check where the check happens? you may need the serverside log.
also what is the server side version you are using?

**Comment by xiaofan-luan** [comment]
![Uploading image.png…]()

**Comment by xiaofan-luan** [comment]
please try 2.3.13. this should not a issue on 2.3.13 or 2.4.1

**Reply by skylersky** [reply]
I changed to 2.4.1, it works now, thank you.
